### PR TITLE
Fix for Issue #907 : Update deprecation warning to use sinon instead of gax.warn

### PR DESCRIPTION
**Pull request by alicejli**
Fix for Issue googleapis/google-cloud-node-core#614 

Update deprecation warning to use sinon stub instead of gax.warn in the test file.

### CODE REVIEW

File: baselines/asset/test/gapic_asset_service_v1.ts.baseline [hunk 1]
```diff
@@ -134,9 +134,18 @@ describe('v1.AssetServiceClient', () => {
...
     });
 
+  const proxyquire = require('proxyquire');
+  const stub = sinon.stub().returns('deprecated');
+  const { AssetServiceClient} = proxyquire(
+    '../src/v1/asset_service_client.js',
+    {
+      'google-gax': {warn: stub},
+    }
+  );
```
Comment (summer-ji-eng) [inline baselines/asset/test/gapic_asset_service_v1.ts.baseline:144]: We need to take care the spaces here to align with up and down.
Ande we should declare the variable under scope of 
```
describe('{{ method.name.toCamelCase() }}', () => {
```
https://github.com/googleapis/gapic-generator-typescript/pull/918/files#diff-d3f0b12ce355b96d095da022dfd7d7cd666455cd78f423616c56a6609d3d8f50R251

**Review by summer-ji-eng — COMMENTED** [review]
Overall looks good me, but since I have bias on stub. Let's wait for Alex to take a look, he might have better way to silent `gax.warn`.

File: baselines/asset/test/gapic_asset_service_v1.ts.baseline (hunk 1 above)
Comment (alicejli) [reply baselines/asset/test/gapic_asset_service_v1.ts.baseline:144]: Thanks for reviewing! Quick question on the scope - right now as it's scope globally in the file, it invokes the stub for all of the tests. If the preference is to be scoped for each set of tests, then I can copy/paste the `proxyquire` code under each set of `describe('')` tests. Is the latter what you're suggesting?

Comment (alicejli) [reply baselines/asset/test/gapic_asset_service_v1.ts.baseline:144]: Okay holding off on this for a little bit as I chatted with Alex offline and we ideally do *not* want to use `proxyquire` as apparently it's a bit unwieldy to use. He's going to investigate how to do this and then I'll update here.

Comment (alicejli) [reply baselines/asset/test/gapic_asset_service_v1.ts.baseline]: Alright! Implemented using `stub` only by adding `warn` as part of the client constructor.

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 2]
```diff
@@ -35,11 +33,7 @@ import { Transform } from 'stream';
 import { RequestType } from 'google-gax/build/src/apitypes';
 {%- endif %}
 import * as protos from '../../protos/protos';
-{%- if api.legacyProtoLoad %}
```
Comment (summer-ji-eng) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk:38]: I guess you didn't have chance rebase on the latest master change. You might need to `git pull origin master` and on your feature branch `git merge master`.

**Review by summer-ji-eng — CHANGES_REQUESTED** [review]
Need to pull the latest code :)

File: baselines/asset/src/v1/asset_service_client.ts.baseline [hunk 3]
```diff
@@ -1033,4 +1037,4 @@ export class AssetServiceClient {
     }
     return Promise.resolve();
   }
-}
+}
```
Comment (alexander-fenster) [inline baselines/asset/src/v1/asset_service_client.ts.baseline:1040]: Looks like the empty line in the end of the template disappeared. Can you please add it back and rerun `npm run baseline`?

File: yarn.lock [hunk 4]
```diff
@@ -3068,4 +3068,4 @@ yn@3.1.1:
 
 yocto-queue@^0.1.0:
   version "0.1.0"
-  resolved "https://registry.npmjs.org/yocto-queue/-/yocto-queue-0.1.0.tgz"
+  resolved "https://registry.npmjs.org/yocto-queue/-/yocto-queue-0.1.0.tgz"
```
Comment (alexander-fenster) [inline yarn.lock:3071]: Please revert this file, its only change is the removed empty line.

**Comment by alicejli** [comment]
Was having a bunch of weird rebasing issues so decided to clean out everything and make a new PR with the changes here: https://github.com/googleapis/gapic-generator-typescript/pull/921.

**Pull request closed without merging**